azext_iot/operations/dps.py

```diff
@@ -558,8 +558,38 @@ def iot_dps_device_enrollment_group_delete(
 
 
 def iot_dps_compute_device_key(
-    cmd, symmetric_key, registration_id,
+    cmd,
+    registration_id,
+    client=None,
+    enrollment_id=None,
+    dps_name=None,
+    resource_group_name=None,
+    symmetric_key=None,
 ):
+    if symmetric_key is None:
+        if not all([client, dps_name, resource_group_name, enrollment_id]):
```

Comment:
When is `client`  `None` and what is the user supposed to do about it?

Reply:
In the original, the client was not passed in at all because no calls were needed to calculate the key offline.

Reply:
Gotcha...to clarify, why is client optional i.e. `client=None` vs required, and related, client is not exposed as a user param correct? Yet it is one of the required elements in the  `if not all()` conditional - so if a user environment has client=None they would trigger the CLIError but there is nothing in the message indicating aht to do in that case.

Reply:
I have changed the client to be required + no longer tracked in the all exist bucket.

To answer your questions:
1. Why is client optional/not optional? The client is only needed for online commands, i.e. getting the dps instance which happens if the user does not provide the symmetric key. The client should be autopopulated as a parameter.
2. If a user's environment has client=None? Not sure what happens but in that scenario other dps commands would be affected too. Client gets populated based on cmd so the user should not be trying to mess with this parameter.

And before you ask: Why did I change the client to be required + removed it from the bucket?
Client is required in other commands so the change is more for consistency.  It should be auto-populated based on the cmd, thus it will have a value no matter what.

Reply:
Thanks. The first line in your reply is what I was driving at, and you made the change. I didn't see the reason to have client be optional when the DPS client is configured to be always passed in (if the respective arg is present) , and have it in the `if not all` check when its not a user stemming issue.